### PR TITLE
fix: sub objects in mongoose are resolved

### PR DESCRIPTION
**Pull request by iamkhalidbashir**
Fixes: #180 

Does not require any new changes to work.

Just use `@FilterableField` decorator with mongoose sub object schemas, and filteration will work

### CODE REVIEW

File: packages/query-graphql/src/types/query/filter.type.ts [hunk 1]
```diff
@@ -103,12 +103,15 @@ function getOrCreateFilterType<T>(
...
     fields.forEach(({ propertyName, target, advancedOptions, returnTypeFunc }) => {
-      const FC = createFilterComparisonType({
-        FieldType: target,
-        fieldName: `${baseName}${upperCaseFirst(propertyName)}`,
-        allowedComparisons: advancedOptions?.allowedComparisons,
-        returnTypeFunc
-      })
+      const objectTypeMetadata = TypeMetadataStorage.getObjectTypeMetadataByTarget(target)
+      const FC = objectTypeMetadata
+        ? getOrCreateFilterType(target, typeName, suffix, depth)
```
Comment (TriPSs) [inline packages/query-graphql/src/types/query/filter.type.ts:108]: Where are these for?

**Review by TriPSs — COMMENTED** [review]
Could you also add some test cases?

File: packages/query-graphql/src/types/query/filter.type.ts (hunk 1 above)
Comment (iamkhalidbashir) [reply packages/query-graphql/src/types/query/filter.type.ts:108]: Basicaly we check here if the DTO class properties type targets are of ObjectType
If they are then we recurrsive call its properties as filter types
(Basically what happens for relations, we do that for fields which are of ObjectType)

**Comment by iamkhalidbashir** [comment]
> Owner

ok I'll try

**Comment by smolinari** [comment]
Any chance you could put the same code into the query-typegoose package?

Scott

**Comment by iamkhalidbashir** [comment]
Never worked with typegoose

**Comment by smolinari** [comment]
> Never worked with typegoose
> […](#)

It's basically the same thing, just much better in terms of TypeScript support. 

Scott

**Comment by iamkhalidbashir** [comment]
@TriPSs I added test cases and still it needs more coverage, could you please look at it as I will be busy now and I need this change to be merged today

**Comment by iamkhalidbashir** [comment]
@smolinari Added support for typegoose

**Comment by smolinari** [comment]
@iamkhalidbashir - Thank you!

Scott

**Comment by TriPSs** [comment]
Thanks, I updated the CI so it always runs everything, coverage is increasing now and nothing failed. I was kinda afraid for some snapshots to fail.

Thanks!

**Pull request merged**